### PR TITLE
proactiveInit wiring in diagnostics

### PR DESCRIPTION
**Pull request by jeet1995**
This PR fills the `proactiveInit` property in the diagnostics. It keeps track of the following information:
- `containers`: The container links of the containers taking part of the connection warm up flow. 
- `pcrc` (proactive connection region count):  The first `k` regions from the `preferredRegions` list to which connections are warmed up.
- `awd` (aggressive warm up duration): The duration in milliseconds within which aggressive connection warm up happens.

```
 "clientCfgs": {
        "id": 1,
        "machineId": "uuid:eb2e9f12-94bf-4ea1-b160-2f59b22cbcba",
        "connectionMode": "DIRECT",
        "numberOfClients": 1,
        "clientEndpoints": {
            "https://localhost:8081/": 1
        },
        "connCfg": {
            "rntbd": "(cto:PT5S, nrto:PT5S, icto:PT0S, ieto:PT1H, mcpe:130, mrpc:30, cer:true)",
            "gw": "(cps:1000, nrto:PT1M, icto:PT1M, p:false)",
            "other": "(ed: true, cs: false, rv: true)"
        },
        "consistencyCfg": "(consistency: null, mm: true, prgns: [eastus,westus])",
        "proactiveInit": "(containers:dbs/RxJava.SDKTest.SharedDatabase_20230726T184149_AZM/colls/id2;dbs/RxJava.SDKTest.SharedDatabase_20230726T184149_AZM/colls/id1;dbs/RxJava.SDKTest.SharedDatabase_20230726T184149_AZM/colls/id0)(pcrc:2)(awd:PT1S)"
    }
```

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfig.java [hunk 1]
```diff
@@ -96,17 +96,19 @@ public String toString() {
...
+                "(containers:%s)(pcrc:%d)(awd:%s)",
                 cosmosContainerIdentities
                     .stream()
                     .map(ci -> String.join(
                         ".",
-                        containerIdAccessor.getDatabaseName(ci),
-                        containerIdAccessor.getContainerName(ci)))
-                    .collect(Collectors.joining(",")),
-                numProactiveConnectionRegions);
+                        containerIdAccessor.getContainerLink(ci)))
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfig.java:108]: why are we using containerLink and not containerName?

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
We should not use containerLink, as it exposes rid of the database, right?

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfig.java (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfig.java:108]: Here `containerLink` refers to `/dbs/{databaseName}/colls/{containerName}` where `databaseName` and `containerName` are supplied by the client application.

**Pull request merged**